react/features/invite/components/PasswordContainer.js

```diff
@@ -0,0 +1,150 @@
+import React, { Component } from 'react';
+
+import { translate } from '../../base/i18n';
+
+import AddPasswordForm from './AddPasswordForm';
+import LockStatePanel from './LockStatePanel';
+import RemovePasswordForm from './RemovePasswordForm';
+
+/**
+ * React component for displaying the current room lock state as well as
+ * exposing features to modify the room lock.
+ */
+class PasswordContainer extends Component {
+    /**
+     * PasswordContainer component's property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * The JitsiConference for which to display a lock state and change the
+         * password.
+         *
+         * @type {JitsiConference}
+         */
+        conference: React.PropTypes.object,
+
+        /**
+         * Whether or not the conference is currently locked.
+         */
+        locked: React.PropTypes.bool,
+
+        /**
+         * Whether or not the room lock, if any, was set by the local user.
+         */
+        lockedLocally: React.PropTypes.bool,
```

Comment:
Ugh, you have to set both `locked` and `lockedLocally`. If you set `lockedLocally` to true, the code will still treat the conference as unlocked. I feel like my idea about an enum fits here so much better.